civihr_employee_portal/views/includes/civihr_employee_portal_handler_absence_table_header_block_area.inc

```diff
@@ -0,0 +1,212 @@
+      $title = $this->applyPlaceholders($alterLink, $absence);
+      $amount = $absence->absence_list_duration / 480;
+
+      $row = array(
+        $title,
+        get_civihr_absence_statuses($absence->absence_list_absence_status),
+      );
+      foreach ($activityAbsenceTypeIds as $typeId) {
+        $row['type_' . $typeId] = '';
+      }
+      $row['type_' . $activityAbsenceTypeIds[$absence->absence_list_activity_type_id]] = ($absence->absence_list_is_credit ? '+ ' : '- ') . $amount;
+      $entitlements[$activityAbsenceTypeIds[$absence->absence_list_activity_type_id]] -= $amount;
+      $rows[] = $row;
+    }
+
+    $rows[] = $this->getBalanceRow($activityAbsenceTypeIds, $entitlements);
+
+    return $rows;
+  }
+
+  /**
+   * Return an array containing balance row.
+   *
+   * @param array $activityAbsenceTypeIds
+   * @param array $balance
+   * @return array
+   */
+  private function getBalanceRow(array $activityAbsenceTypeIds, array $balance) {
+
+    $row = array(
+      t('Balance') . ':',
+      '',
+    );
+    foreach ($activityAbsenceTypeIds as $activityId => $typeId) {
+      $row['type_' . $typeId] = $balance[$typeId];
+    }
+
+    return $row;
+  }
+
+  /**
+   * Return an array containing Absence types, excluding 'Sick' type.
+   *
+   * @return array
+   */
+  private function getAbsenceTypes() {
+
+    $result = array();
+    $absenceTypes = get_civihr_absence_types();
+
+    foreach ($absenceTypes as $absenceType) {
+      if (civihr_employee_portal_is_sick_absence($absenceType['name'])) {
+        continue;
+      }
+      $result[] = $absenceType;
+    }
+
+    return $result;
+  }
+
+  /**
+   * Return an array containing key / value pairs of activity type id / absence
+   * type id. Used for mapping activity types to their absence types equivalents.
+   *
+   * @return array
+   */
+  private function getActivityAbsenceTypeIds() {
+
+    $result = array();
+    $absenceTypes = $this->getAbsenceTypes();
+
+    foreach ($absenceTypes as $type) {
+      if (!empty($type['debit_activity_type_id'])) {
+        $result[$type['debit_activity_type_id']] = $type['id'];
+      }
+      if (!empty($type['credit_activity_type_id'])) {
+        $result[$type['credit_activity_type_id']] = $type['id'];
+      }
+    }
+
+    return $result;
+  }
+
+  /**
+   * Return an assotiative array containing key / value pairs of
+   * absence_type / amount values of currently logged in user for
+   * given period.
+   *
+   * @param int $periodId
+   * @return array
+   */
+  private function getEntitlementData($periodId) {
+
+    $entitlementData = array();
+    $query = db_select('civicrm_hrabsence_entitlement', 'che')
+      ->fields('che', array('type_id', 'amount'))
+      ->orderBy('che.contact_id');
+
+    $query->where('(che.contact_id = :c_id)', array(':c_id' => $_SESSION['CiviCRM']['userID']));
+    $query->where('(che.period_id = :period_id)', array(':period_id' => $periodId));
+    $query->where('(che.type_id <> 1)');
```

Comment:
Yeah, I've changed the method to use API anyway (like the other methods) and added a helper function to SSP module which returns 'sick' type ID.